lib/active_record/typed_store/dsl.rb

```diff
@@ -9,36 +9,18 @@ class DSL
     def initialize(store_name, options)
       @coder = options.fetch(:coder) { default_coder(store_name) }
       @store_name = store_name
-      @prefix =
-        case options[:prefix]
-        when String, Symbol
-          "#{options[:prefix]}_"
-        when true
-          "#{store_name}_"
-        when false, nil
-          ""
-        else
-          raise ArgumentError, "Unexpected type for prefix option. Expected string, symbol, or boolean"
-        end
-      @suffix =
-        case options[:suffix]
-        when String, Symbol
-          "_#{options[:suffix]}"
-        when true
-          "_#{store_name}"
-        when false, nil
-          ""
-        else
-          raise ArgumentError, "Unexpected type for suffix option. Expected string, symbol, or boolean"
-        end
-      @accessors = if options[:accessors] == false
-        {}
-      elsif options[:accessors].is_a?(Array)
-        options[:accessors].each_with_object({}) do |accessor_name, hash|
-          hash[accessor_name] = accessor_key_for(accessor_name)
-        end
-      end
       @fields = {}
+    end
+
+    def store_accessors(options)
+      set_affixes(options)
+      @accessors = if options[:accessors] == false
```

Comment:
I've let the accessors be for now, but I think those should also be fixed so they are preserved.